llvm/lib/ObjCopy/ELF/ELFObject.cpp

```diff
@@ -2847,24 +2847,24 @@ void SRECSectionWriter::writeRecord(SRecord &Record, uint64_t Off) {
   memcpy(Out.getBufferStart() + Off, Data.data(), Data.size());
 }
 
-void SRECSectionWriter::writeRecords(uint32_t Entry) {
+void SRECSizeCalculator::writeRecords(uint32_t Entry) {
```

Comment:
Taking another look at this and it's not really ideal that we have to do two passes through the records (one for sizing and one for writing). I'm sure there's a reason to do it this way, but what is it, as that might inform an alternative approach?

Reply:
We can write to the output buffer without resizing or copying.